lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -246,31 +247,41 @@ sub _get_dirty_status {
 }
 
 # Obs version is parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
-sub _get_version_in_folder {
+sub _get_builds_in_folder {
     my ($folder) = @_;
     open(my $fh, '<', Mojo::File->new($folder, $files_iso_filename)) or return "";
-    my $version;
+    my %seen;
     while (my $row = <$fh>) {
         chomp $row;
         next unless $row;
-        next if substr($row, 0, 1) eq "#";
-        if ($row =~ m/Build((\d)+\.(\d)+(\.(\d)+)?)/ or $row =~ m/Snapshot((\d)+(\.(\d)+)*)/) {
-            $version = $1;
-            last;
-        }
+        next unless ($row =~ m/Build((\d)+\.(\d)+(\.(\d)+)?)/ or $row =~ m/Snapshot((\d)+(\.(\d)+)*)/);
+        my $build = $1;
+        $seen{$build} = 1;
     }
     close $fh;
-    return $version;
+    return sort { $b cmp $a } keys %seen;
 }
 
-# Obs version is parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
-sub _get_obs_version {
-    my ($c, $project) = @_;
-    my $helper = $c->obs_rsync;
-    my $home   = $helper->home;
-    (my $batch, $project) = $helper->get_first_batch($project);
+# Obs builds are parsed from files_iso.lst, which is updated from ObsRsync Gru tasks
+sub _get_obs_builds_text {
+    my ($c, $project, $last) = @_;
+    my $helper    = $c->obs_rsync;
+    my $home      = $helper->home;
+    my $subfolder = $last ? '.run_last' : '';
+    my %seen;
+    my $sub = sub {
+        my ($_project, $_batch) = @_;
+        my @_builds = _get_builds_in_folder(Mojo::File->new($home, $_project, $_batch, $subfolder));
+        for my $build (@_builds) {
+            $seen{$build} = 1 if $build;
+        }
+        return;
```

Comment:
We don't use bare returns, since they are context sensitive, which is a common cause for security issues.